python/cugraph/cugraph/tests/internals/test_replicate_edgelist_mg.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (c) 2022-2023, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import gc
+
+import pytest
+
+# import pandas as pd
+
+import dask_cudf
+import numpy as np
+import cugraph
+from cugraph.testing import UNDIRECTED_DATASETS, karate_disjoint
+
+# from cugraph.structure.replicate_edgelist import replicate_edgelist
+from cudf.testing.testing import assert_frame_equal
+from pylibcugraph.testing.utils import gen_fixture_params_product
+
+
+# =============================================================================
+# Pytest Setup / Teardown - called for each test function
+# =============================================================================
+def setup_function():
+    gc.collect()
+
+
+edgeWeightCol = "weights"
+edgeIdCol = "edge_id"
+edgeTypeCol = "edge_type"
+srcCol = "src"
+dstCol = "dst"
+
+
+input_data = UNDIRECTED_DATASETS
+input_data.append(karate_disjoint)
+print("type UNDIRECTED_DATASETS = ", type(UNDIRECTED_DATASETS))
+print("input_data = ", input_data)
+datasets = [pytest.param(d) for d in input_data]
+
+fixture_params = gen_fixture_params_product(
+    (datasets, "graph_file"),
+    ([True, False], "distributed"),
+    ([True, False], "use_weights"),
+    ([True, False], "use_edge_ids"),
+    ([True, False], "use_edge_type_ids"),
+)
+
+
+@pytest.fixture(scope="module", params=fixture_params)
+def input_combo(request):
+    """
+    Simply return the current combination of params as a dictionary for use in
+    tests or other parameterized fixtures.
+    """
+    return dict(
+        zip(
+            (
+                "graph_file",
+                "use_weights",
+                "use_edge_ids",
+                "use_edge_type_ids",
+                "distributed",
+            ),
+            request.param,
+        )
+    )
```

Comment:
This pattern isn't as useful here since there's only one test.  It might be better for future maintainers if the test in this file just used regular pytest parameters.